lib/cache/cache.go

```diff
@@ -1051,20 +1047,18 @@ func (c *Cache) performRelativeNodeExpiry(ctx context.Context) error {
 			continue
 		}
 
-		// event stream processing is paused while this function runs.  we perform the
-		// actual expiry by constructing a fake delete event for the resource which both
-		// updates this cache, and all downstream caches.
-		err = c.processEvent(ctx, types.Event{
-			Type: types.OpDelete,
-			Resource: &types.ResourceHeader{
-				Kind:     types.KindNode,
-				Metadata: node.GetMetadata(),
-			},
-		})
-		if err != nil {
+		// remove the node locally without emitting an event, other caches will
+		// eventually remove the same node when they run their expiry logic.
+		if err := c.presenceCache.DeleteNode(ctx, apidefaults.Namespace, node.GetName()); err != nil {
 			return trace.Wrap(err)
 		}
```

Comment:
Nit: Instead of directly calling `DeleteNode`, how about amending `processEvent` with an option that just skips emission/fanout?  That way we still have a "single source of truth" for how nodes get removed (i.e. node removal is always performed by the removal logic in `collections.go` regardless of whether this is a real expiry or a relative expiry).  Minor diff, but IMO less brittle in the event we ever refactor how node deletion is handled, or decide to inject stats/etc in the future.